libsql-server/src/schema/scheduler.rs

```diff
@@ -523,6 +531,10 @@ async fn try_step_task_inner(
             let (new_status, error) = step_task(&mut txn, job_id)?;
             txn.commit()?;
 
```

Comment:
There's a bunch of statements here that can return with error and we won't re-enable foreign_keys then, no?